powershell-adapter/psDscAdapter/psDscAdapter.psm1

```diff
@@ -37,99 +50,188 @@ function Invoke-DscCacheRefresh {
         $Module
     )
 
-    # create a list object to store cache of Get-DscResource
-    [dscResourceCache[]]$dscResourceCache = [System.Collections.Generic.List[Object]]::new()
+    $refreshCache = $false
+
+    $cacheFilePath = if ($IsWindows) {
+        # PS 6+ on Windows
+        Join-Path $env:LocalAppData "dsc\PSAdapterCache.json"
+    } else {
+        # either WinPS or PS 6+ on Linux/Mac
+        if ($PSVersionTable.PSVersion.Major -le 5) {
+            Join-Path $env:LocalAppData "dsc\WindowsPSAdapterCache.json"
```

Comment:
Doesn't this need to be in the `if ($IsWindows)` block?

Reply:
Windows PS does not know "$IsWindows" env var; it evaluates to $null there.